_posts/2018-05-07-A-Machine-Learning-Example-For-Business.md

```diff
@@ -0,0 +1,271 @@
+
+The point is not to learn inside out what these numbers mean, but to understand that this is the <i>model</i> and the software understands how to apply it for the predictions:
+```R
+predictions <- predict(model, newdata=data.test)
+```
+
+By doing some copy and paste operations, the predictions can be appended to testing data:
+
+![Naive bayes predictions](/img/a-machine-learning-example-for-business/naive-bayes-predictions.png)
+
+If the value in `Yes` column is more than 0.5, it means that the prediction for that order is to arrive late.
+
+<blockquote>
+In our case the order of records matter: The model should be trained with observations that have occurred before the observations in test data. The number of week (`week`) is a helper column to order the data for train and test splitting.
+</blockquote>
+
+## Evaluating the goodness of the model
+A good way to evaluate the performance of the model in this example is to calculate how many predictions we got correct. The actual value can be seen from `is_late` column and the prediction from `Prediction` column.
+
+Because I had the power to create the data for this example, all 6 predictions were correct. For the second record the probability for both being late and not being late was 50%. If this would have been interpreted as `Yes`, the prediction would have been incorrect, and the accuracy would have fallen to 83.3% or 5/6.
+
+![Naive bayes prediction 50-50](/img/a-machine-learning-example-for-business/naive-bayes-prediction-50-50.png)
+
+## Evaluating the business benefit
+So far we have:
+1. Identified a business problem.
+2. Combined and cleaned possibly relevant data.
+3. Selected a suitable algorithm.
+4. Trained a model.
+5. Tested the model with historical data.
+6. Evaluated the prediction accuracy.
+
+Now it's time to evaluate the business benefit.
+
+We are confident, that our model is between 83.3% to 100% accurate for unseen observations, where only order day, order size and sub contractor are known before setting up the order. Let's trust the mid point and say that accuracy is 91.7%.
+
+You can count that in test data 3/6 or 50% of orders arrived late. Supposing that you would have used your trained machine learning model, you could have made the order only for those 3 or 4 deliveries that are predicted to arrive on time and a couple of additional orders that are not described in the test data.
+
+In real world the situation would be more complex, but you get the idea that accuracy of 91.7% is more than 50%. With these numbers you are probably also able to estimate the impact in terms of money.
+
+## Implementing the model to daily work
+
+Here are some ways to harness the solution for business.
+
+<b>Data study</b>. Improve company's process by experimenting the optimal order types.
+
+<b>Calculator</b>. Let employers to enter the three predictor variables to a web calculator before placing the order to evaluate the risk of receiving products late.
```

Comment:
"Let employers enter..." or "Allow employers to enter..." sounds better to my ear.

Reply:
Ok.